lib/active_merchant/billing/gateways/fortis.rb

```diff
@@ -175,11 +182,15 @@ def path(action, value = '')
           credit: '/transactions/cc/refund/keyed',
           store: '/tokens/cc',
           unstore: '/tokens/{placeholder}'
-        }[action]&.gsub('{placeholder}', value.to_s)
+        }[action]&.gsub('{placeholder}', value.to_s.split('|').first || '')
       end
 
       def payment_type(payment)
-        payment.is_a?(String) ? 'token' : 'keyed'
+        if payment.is_a?(String)
+          payment.split('|').last == 'txn' ? 'prev-trxn' : 'token'
```

Comment:
it took me a little bit to understand why the split is needed, but I can see that it is required for case like this. `"transaction/31e59fb2ec9d4bc5fb|txn/refund"` 👍🏼